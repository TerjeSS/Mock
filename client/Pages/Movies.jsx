import useLoading from "../Components/useLoading";

export default function Movies({ listMovies }) {
  const { loading, data, error } = useLoading(listMovies);
  if (loading) {
    return <div>Loading.....</div>;
  }
  if (error) {
    return <div>{error.toString()}</div>;
  }
  return (
    <div>
      <h1>List of movies</h1>
      {data.map((m) => {
        return (
          <div>
            <h1>{m.name}</h1>
            <h3>Released: {m.released}</h3>
            <p>{m.plot}</p>
          </div>
        );
      })}
    </div>
  );
}
