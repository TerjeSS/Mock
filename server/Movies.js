export default function Movies() {
  return [
    {
      name: "Movie nr 1",
      released: "13531431123",
      plot: "falfdlks askdfjqeijf fasdljkfjsafewiåp fslakfjaski  fdkslafjlsødkfjadslkfjasldø ",
    },
    {
      name: "Movie nr 2",
      released: "13531431123",
      plot: "falfdlks askdfjqeijf fasdljkfjsafewiåp fslakfjaski  fdkslafjlsødkfjadslkfjasldø ",
    },
    {
      name: "Movie nr 3",
      released: "13531431123",
      plot: "falfdlks askdfjqeijf fasdljkfjsafewiåp fslakfjaski  fdkslafjlsødkfjadslkfjasldø ",
    },
    {
      name: "Movie nr 4",
      released: "13531431123",
      plot: "falfdlks askdfjqeijf fasdljkfjsafewiåp fslakfjaski  fdkslafjlsødkfjadslkfjasldø ",
    },
  ];
}
