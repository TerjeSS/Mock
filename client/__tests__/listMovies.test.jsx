import Movies from "../Pages/Movies";
import ReactDOM from "react-dom";
import { act } from "react-dom/test-utils";
import * as React from "react";

describe("Movies component", () => {
  it("shows loading screen", () => {
    const domElement = document.createElement("div");
    ReactDOM.render(<Movies />, domElement);
    expect(domElement.innerHTML).toMatchSnapshot();
  });

  it("should show movies", async () => {
    const movies = [{ name: "movie 1" }, { name: "movie 2" }];
    const domElement = document.createElement("div");
    await act(async () => {
      ReactDOM.render(<Movies listMovies={() => movies} />, domElement);
    });
    expect(
      Array.from(domElement.querySelectorAll("h2")).map((e) => e.innerHTML)
    ).toEqual(["movie 1", "movie 2"]);

    expect(domElement.innerHTML).toMatchSnapshot();
  });

  it("should display error", async () => {
    const domElement = document.createElement("div");
    await act(async () => {
      ReactDOM.render(
        <Movies
          listMovies={() => {
            throw new Error("Something went wrong");
          }}
        />,
        domElement
      );
    });
    expect(domElement.querySelector("#error-text").innerHTML).toEqual(
      "Error: Something went wrong"
    );
  });
});
