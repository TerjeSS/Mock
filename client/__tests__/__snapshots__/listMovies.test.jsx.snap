// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies component should show movies 1`] = `"<div><h1>List of movies</h1><div><h2>movie 1</h2><h3>Released: </h3><p></p></div><div><h2>movie 2</h2><h3>Released: </h3><p></p></div></div>"`;

exports[`Movies component shows loading screen 1`] = `"<div>Loading.....</div>"`;
